src/main/scala/cognite/spark/v1/NumericDataPointsRdd.scala

```diff
@@ -185,9 +185,12 @@ final case class NumericDataPointsRdd(
       id: CogniteId,
       start: Instant,
       end: Instant,
-      granularities: Seq[Granularity] = granularitiesToTry): IO[Option[Seq[Range]]] =
+      granularities: List[Granularity] = granularitiesToTry): IO[Option[Seq[Range]]] =
     granularities match {
-      case granularity +: moreGranular =>
+      case Nil =>
+        throw new RuntimeException(
+          "No granularities requested for data points aggregation. Please report this to support@cognite.com")
```

Comment:
I wonder if we haven't run into an issue when granularities before 😱 , anyway this would help to trace back the error easily 😁

Reply:
We're the only callers of this (private) method, so it is (or should be) impossible. Arguably we should have used a `NonEmptyList` to have the compiler verify it for us, but this is a simple fix to tell the compiler that we think we know what we're doing, and to tell users to inform us if it turns out we do not know what we're doing. :)